fbw-a380x/src/systems/instruments/src/MFD/pages/FMS/F-PLN/MfdFmsFplnArr.tsx

```diff
@@ -394,6 +423,21 @@ export class MfdFmsFplnArr extends FmsPage<MfdFmsFplnArrProps> {
         }
       }, true),
     );
+
+    NXDataStore.getAndSubscribe('CONFIG_USING_METRIC_UNIT', (key, value) => {
+      value === '1' ? this.unit.set(UnitType.METER) : this.unit.set(UnitType.FOOT);
+
+      if (!this.props.fmcService.master || !this.loadedFlightPlan) {
+        return;
+      }
+
+      const isAltn = this.props.fmcService.master.revisedWaypointIsAltn.get() ?? false;
+      const flightPlan = isAltn ? this.loadedFlightPlan.alternateFlightPlan : this.loadedFlightPlan;
+
+      if (flightPlan.destinationAirport) {
+        this.GenerateRunwayOptions(flightPlan, isAltn);
+      }
+    });
```

Comment:
I'd do all the other work in a sub on lengthUnit for cleanliness. METER needs to be the default, both for invalid garbage in the setting, hence inverting the condition, and if it's not set at all (the 3rd param for getAndSubscribe).
```suggestion
    NXDataStore.getAndSubscribe('CONFIG_USING_METRIC_UNIT', (key, value) => {
      this.lengthUnit.set(value === '0' ? UnitType.FOOT : UnitType.METER);
    }, '1');
      
    this.lengthUnit.sub(() => {
      if (!this.props.fmcService.master || !this.loadedFlightPlan) {
        return;
      }

      const isAltn = this.props.fmcService.master.revisedWaypointIsAltn.get() ?? false;
      const flightPlan = isAltn ? this.loadedFlightPlan.alternateFlightPlan : this.loadedFlightPlan;

      if (flightPlan.destinationAirport) {
        this.GenerateRunwayOptions(flightPlan, isAltn);
      }
    }, true);
```